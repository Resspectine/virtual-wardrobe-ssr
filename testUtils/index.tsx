import { queries, Queries, render as reactRender, RenderOptions, RenderResult } from '@testing-library/react';
import { FC, ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';

import { ThemeProvider } from '@/materialUiTheme/provider';

export const sleep = (timeout: number): Promise<void> =>
  new Promise(resolve => {
    setTimeout(() => {
      resolve();
    }, timeout);
  });

export const render = <Q extends Queries = typeof queries, Container extends Element | DocumentFragment = HTMLElement>(
  ui: ReactElement,
  options: RenderOptions<Q, Container> = {}
): RenderResult<Q, Container> => {
  const queryClient = new QueryClient();

  const Wrapper: FC = ({ children }) => (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider>{children}</ThemeProvider>
    </QueryClientProvider>
  );

  return { ...reactRender(ui, { ...options, wrapper: Wrapper }) };
};
