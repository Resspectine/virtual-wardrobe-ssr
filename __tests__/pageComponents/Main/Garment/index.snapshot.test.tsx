import GarmentComponent from '@/pageComponents/Main/Garment';
import { render } from '@/testUtils';

const cb = () => {};

describe('Garment', () => {
  test('should match snapshot not favorite state', async () => {
    const { asFragment } = render(
      <GarmentComponent
        garment={{
          description: '',
          id: '',
          isFavorite: false,
          price: '',
          title: '',
          wearingAmount: 0,
          tags: [
            {
              id: '1',
              title: '',
            },
          ],
        }}
        onClick={cb}
        toggleFavorite={cb}
        onMouseDown={cb}
        onMouseUp={cb}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('should match snapshot favorite state', async () => {
    const { asFragment } = render(
      <GarmentComponent
        garment={{
          description: '',
          id: '',
          isFavorite: true,
          price: '',
          title: '',
          wearingAmount: 0,
          tags: [
            {
              id: '1',
              title: '',
            },
          ],
        }}
        onClick={cb}
        toggleFavorite={cb}
        onMouseDown={cb}
        onMouseUp={cb}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
