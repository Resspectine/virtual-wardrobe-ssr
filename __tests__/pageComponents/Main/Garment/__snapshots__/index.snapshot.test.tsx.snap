// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Garment should match snapshot favorite state 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  padding: 10px;
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  position: relative;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  box-shadow: 0px 3px 5px -1px rgb(0 0 0 / 20%),0px 5px 8px 0px rgb(0 0 0 / 14%),0px 1px 14px 0px rgb(0 0 0 / 12%);
}

.emotion-1 {
  position: absolute;
  top: 10px;
  right: 10px;
  background-color: #388e3c;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px;
  border-radius: 8px;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  fill: #fff;
}

.emotion-5 {
  margin: 0;
  font-family: Montserrat,Times New Roman;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 20px;
  line-height: 20px;
  margin-bottom: 16px;
  font-weight: 500;
}

.emotion-6 {
  margin: 0;
  font-family: Montserrat,Times New Roman;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 16px;
  line-height: 16px;
  margin-bottom: 16px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -5px;
  margin-right: -5px;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-8 {
  max-height: 40px;
  padding: 5px 16px;
  background-color: #f5f5f5;
  border-radius: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: default;
  margin: 0px 5px;
  margin-bottom: 10px;
}

.emotion-9 {
  margin: 0;
  font-family: Montserrat,Times New Roman;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 14px;
  line-height: 14px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-11 {
  margin: 0;
  font-family: Montserrat,Times New Roman;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 18px;
  line-height: 18px;
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-12 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-13 {
  margin: 0;
  font-family: Montserrat,Times New Roman;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 18px;
  line-height: 18px;
  font-weight: 500;
}

<div
    class="MuiBox-root emotion-0"
    data-testid="garment"
  >
    <div
      aria-label="Remove from favorite"
      class=" MuiBox-root emotion-1"
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
        data-testid="StarBorderIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
        />
      </svg>
    </div>
    <div
      class="MuiBox-root emotion-3"
    >
      <div
        class="MuiBox-root emotion-3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-5"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-6"
        />
      </div>
    </div>
    <div
      class="MuiBox-root emotion-7"
    >
      <div
        class="MuiBox-root emotion-8"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-9"
        />
      </div>
    </div>
    <div
      class="MuiBox-root emotion-10"
    >
      <p
        aria-label="The number of times you have worn this garment. Double click on garment to wear"
        class="MuiTypography-root MuiTypography-body1  emotion-11"
        data-mui-internal-clone-element="true"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-12"
          data-testid="GestureIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4.59 6.89c.7-.71 1.4-1.35 1.71-1.22.5.2 0 1.03-.3 1.52-.25.42-2.86 3.89-2.86 6.31 0 1.28.48 2.34 1.34 2.98.75.56 1.74.73 2.64.46 1.07-.31 1.95-1.4 3.06-2.77 1.21-1.49 2.83-3.44 4.08-3.44 1.63 0 1.65 1.01 1.76 1.79-3.78.64-5.38 3.67-5.38 5.37 0 1.7 1.44 3.09 3.21 3.09 1.63 0 4.29-1.33 4.69-6.1H21v-2.5h-2.47c-.15-1.65-1.09-4.2-4.03-4.2-2.25 0-4.18 1.91-4.94 2.84-.58.73-2.06 2.48-2.29 2.72-.25.3-.68.84-1.11.84-.45 0-.72-.83-.36-1.92.35-1.09 1.4-2.86 1.85-3.52.78-1.14 1.3-1.92 1.3-3.28C8.95 3.69 7.31 3 6.44 3 5.12 3 3.97 4 3.72 4.25c-.36.36-.66.66-.88.93l1.75 1.71zm9.29 11.66c-.31 0-.74-.26-.74-.72 0-.6.73-2.2 2.87-2.76-.3 2.69-1.43 3.48-2.13 3.48z"
          />
        </svg>
        0
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 emotion-13"
      >
        $
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Garment should match snapshot not favorite state 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  padding: 10px;
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  position: relative;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  box-shadow: 0px 3px 5px -1px rgb(0 0 0 / 20%),0px 5px 8px 0px rgb(0 0 0 / 14%),0px 1px 14px 0px rgb(0 0 0 / 12%);
}

.emotion-1 {
  position: absolute;
  top: 10px;
  right: 10px;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px;
  border-radius: 8px;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  fill: initial;
}

.emotion-5 {
  margin: 0;
  font-family: Montserrat,Times New Roman;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 20px;
  line-height: 20px;
  margin-bottom: 16px;
  font-weight: 500;
}

.emotion-6 {
  margin: 0;
  font-family: Montserrat,Times New Roman;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 16px;
  line-height: 16px;
  margin-bottom: 16px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -5px;
  margin-right: -5px;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-8 {
  max-height: 40px;
  padding: 5px 16px;
  background-color: #f5f5f5;
  border-radius: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: default;
  margin: 0px 5px;
  margin-bottom: 10px;
}

.emotion-9 {
  margin: 0;
  font-family: Montserrat,Times New Roman;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 14px;
  line-height: 14px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-11 {
  margin: 0;
  font-family: Montserrat,Times New Roman;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 18px;
  line-height: 18px;
  font-weight: 500;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-12 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-13 {
  margin: 0;
  font-family: Montserrat,Times New Roman;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 18px;
  line-height: 18px;
  font-weight: 500;
}

<div
    class="MuiBox-root emotion-0"
    data-testid="garment"
  >
    <div
      aria-label="Add to favorite"
      class=" MuiBox-root emotion-1"
      data-mui-internal-clone-element="true"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-2"
        data-testid="StarBorderIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
        />
      </svg>
    </div>
    <div
      class="MuiBox-root emotion-3"
    >
      <div
        class="MuiBox-root emotion-3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-5"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-6"
        />
      </div>
    </div>
    <div
      class="MuiBox-root emotion-7"
    >
      <div
        class="MuiBox-root emotion-8"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-9"
        />
      </div>
    </div>
    <div
      class="MuiBox-root emotion-10"
    >
      <p
        aria-label="The number of times you have worn this garment. Double click on garment to wear"
        class="MuiTypography-root MuiTypography-body1  emotion-11"
        data-mui-internal-clone-element="true"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-12"
          data-testid="GestureIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4.59 6.89c.7-.71 1.4-1.35 1.71-1.22.5.2 0 1.03-.3 1.52-.25.42-2.86 3.89-2.86 6.31 0 1.28.48 2.34 1.34 2.98.75.56 1.74.73 2.64.46 1.07-.31 1.95-1.4 3.06-2.77 1.21-1.49 2.83-3.44 4.08-3.44 1.63 0 1.65 1.01 1.76 1.79-3.78.64-5.38 3.67-5.38 5.37 0 1.7 1.44 3.09 3.21 3.09 1.63 0 4.29-1.33 4.69-6.1H21v-2.5h-2.47c-.15-1.65-1.09-4.2-4.03-4.2-2.25 0-4.18 1.91-4.94 2.84-.58.73-2.06 2.48-2.29 2.72-.25.3-.68.84-1.11.84-.45 0-.72-.83-.36-1.92.35-1.09 1.4-2.86 1.85-3.52.78-1.14 1.3-1.92 1.3-3.28C8.95 3.69 7.31 3 6.44 3 5.12 3 3.97 4 3.72 4.25c-.36.36-.66.66-.88.93l1.75 1.71zm9.29 11.66c-.31 0-.74-.26-.74-.72 0-.6.73-2.2 2.87-2.76-.3 2.69-1.43 3.48-2.13 3.48z"
          />
        </svg>
        0
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 emotion-13"
      >
        $
      </p>
    </div>
  </div>
</DocumentFragment>
`;
